build/lib/pgBackRestBuild/Config/Data.pm

```diff
@@ -0,0 +1,2104 @@
+        &CFGBLDDEF_RULE_TYPE => CFGBLDDEF_TYPE_BOOLEAN,
+        &CFGBLDDEF_RULE_DEFAULT => false,
+        &CFGBLDDEF_RULE_COMMAND =>
+        {
+            &CFGBLDCMD_BACKUP => {},
+        }
+    },
+
+    &CFGBLDOPT_MANIFEST_SAVE_THRESHOLD =>
+    {
+        &CFGBLDDEF_RULE_SECTION => CFGBLDDEF_SECTION_GLOBAL,
+        &CFGBLDDEF_RULE_TYPE => CFGBLDDEF_TYPE_INTEGER,
+        &CFGBLDDEF_RULE_DEFAULT => 1073741824,
+        &CFGBLDDEF_RULE_COMMAND =>
+        {
+            &CFGBLDCMD_BACKUP => {},
+        }
+    },
+
+    &CFGBLDOPT_RESUME =>
+    {
+        &CFGBLDDEF_RULE_SECTION => CFGBLDDEF_SECTION_GLOBAL,
+        &CFGBLDDEF_RULE_TYPE => CFGBLDDEF_TYPE_BOOLEAN,
+        &CFGBLDDEF_RULE_DEFAULT => true,
+        &CFGBLDDEF_RULE_COMMAND =>
+        {
+            &CFGBLDCMD_BACKUP => {},
+        }
+    },
+
+    &CFGBLDOPT_START_FAST =>
+    {
+        &CFGBLDDEF_RULE_SECTION => CFGBLDDEF_SECTION_GLOBAL,
+        &CFGBLDDEF_RULE_TYPE => CFGBLDDEF_TYPE_BOOLEAN,
+        &CFGBLDDEF_RULE_DEFAULT => false,
+        &CFGBLDDEF_RULE_COMMAND =>
+        {
+            &CFGBLDCMD_BACKUP => {},
+        }
+    },
+
+    &CFGBLDOPT_STOP_AUTO =>
+    {
+        &CFGBLDDEF_RULE_SECTION => CFGBLDDEF_SECTION_GLOBAL,
+        &CFGBLDDEF_RULE_TYPE => CFGBLDDEF_TYPE_BOOLEAN,
+        &CFGBLDDEF_RULE_DEFAULT => false,
+        &CFGBLDDEF_RULE_COMMAND =>
+        {
+            &CFGBLDCMD_BACKUP => {},
+        }
+    },
+
+    # Expire options
+    #-------------------------------------------------------------------------------------------------------------------------------
+    &CFGBLDOPT_RETENTION_ARCHIVE =>
+    {
+        &CFGBLDDEF_RULE_SECTION => CFGBLDDEF_SECTION_GLOBAL,
+        &CFGBLDDEF_RULE_TYPE => CFGBLDDEF_TYPE_INTEGER,
+        &CFGBLDDEF_RULE_REQUIRED => false,
+        &CFGBLDDEF_RULE_ALLOW_RANGE => [CFGBLDDEF_DEFAULT_RETENTION_MIN, CFGBLDDEF_DEFAULT_RETENTION_MAX],
+        &CFGBLDDEF_RULE_COMMAND =>
+        {
+            &CFGBLDCMD_BACKUP => {},
+            &CFGBLDCMD_EXPIRE => {},
+        }
+    },
+
+    &CFGBLDOPT_RETENTION_ARCHIVE_TYPE =>
+    {
+        &CFGBLDDEF_RULE_SECTION => CFGBLDDEF_SECTION_GLOBAL,
+        &CFGBLDDEF_RULE_TYPE => CFGBLDDEF_TYPE_STRING,
+        &CFGBLDDEF_RULE_DEFAULT => CFGBLDOPTVAL_BACKUP_TYPE_FULL,
+        &CFGBLDDEF_RULE_COMMAND =>
+        {
+            &CFGBLDCMD_BACKUP => {},
+            &CFGBLDCMD_EXPIRE => {},
+        },
+        &CFGBLDDEF_RULE_ALLOW_LIST =>
+        {
+            &CFGBLDOPTVAL_BACKUP_TYPE_FULL => 1,
+            &CFGBLDOPTVAL_BACKUP_TYPE_DIFF => 1,
+            &CFGBLDOPTVAL_BACKUP_TYPE_INCR => 1
+        }
+    },
+
+    &CFGBLDOPT_RETENTION_DIFF =>
+    {
+        &CFGBLDDEF_RULE_SECTION => CFGBLDDEF_SECTION_GLOBAL,
+        &CFGBLDDEF_RULE_TYPE => CFGBLDDEF_TYPE_INTEGER,
+        &CFGBLDDEF_RULE_REQUIRED => false,
+        &CFGBLDDEF_RULE_ALLOW_RANGE => [CFGBLDDEF_DEFAULT_RETENTION_MIN, CFGBLDDEF_DEFAULT_RETENTION_MAX],
+        
```

Comment:
It is a minimal change to remove it for this commit (I already did it to make sure it worked). I found it confusing to try to understand why the s3 stuff was not "typed". I find it much cleaner to see it displayed in the hash than buried in default logic. Stuff that's used internally (like the other defaults you add) is fine, but things that I and a user who needs to add to the config file - like is this an integer? a string? a what? I feel should be clearly and consistently laid out.